Lib/pathlib.py

```diff
@@ -1264,13 +1264,15 @@ def write_bytes(self, data):
         with self.open(mode='wb') as f:
             return f.write(view)
 
-    def write_text(self, data, encoding=None, errors=None):
+    def write_text(self, data, encoding=None, errors=None, newline=None):
         """
         Open the file in text mode, write to it, and close the file.
         """
         if not isinstance(data, str):
             raise TypeError('data must be str, not %s' %
                             data.__class__.__name__)
+        if newline is not None:
+            data = re.sub('\r\n|\n|\r', newline, data)
```

Comment:
Why not just pass the `newline` argument to `open()`?

Reply:
`Path.open()` does not support any sequence to be newline, only `None, '', '\n', '\r', and '\r\n'`.

Reply:
Fixed, changed `re.sub()` to passing newline parameter to `open()`.